runtime-config-linux.md

```diff
@@ -144,19 +144,29 @@ Also known as cgroups, they are used to restrict resource usage for a container
 cgroups provide controls to restrict cpu, memory, IO, pids and network for the container.
 For more information, see the [kernel cgroups documentation](https://www.kernel.org/doc/Documentation/cgroups/cgroups.txt).
 
-The path to the cgroups can be specified in the Spec via `cgroupsPath`.
-`cgroupsPath` is expected to be relative to the cgroups mount point.
-If `cgroupsPath` is not specified, implementations can define the default cgroup path.
+The name of the cgroups can be specified in the Spec via `cgroupsName`.
+Cgroup names mimic filesystem paths closely since they express a hierarchy of cgroups.
+`cgroupsName` is expected to be absolute.
+An absolute name is one that is defined from the root cgroup `/`.
+For example, `/foo/bar` is acceptible. `foo/bar` is not acceptible.
+Allowable characters for cgroup names are,
+Alpha numeric ([a-zA-Z0-9]+)
+Underscores (_)
+Dashes (-)
+Periods (.)
+In general and unless otherwise specified, regular filesystem path rules apply.
+
+If `cgroupsName` is not specified, implementations can choose to use (or not use) any cgroups.
```

Comment:
On Wed, Dec 16, 2015 at 12:57:42PM -0800, Mrunal Patel wrote:
> > +If `cgroupsName` is not specified, implementations can choose to use (or not use) any cgroups.
> 
> @wking IMO, it should mean pick their own default. We need a way to
> say don't do anything and for that resources == nil and cgroupsName
> == "" should work.

Or cgroupsName = nil to avoid needing to qualify the slash-start
requirement (see also *string discussion in #233), but yeah.

For the requirements != nil && cgroupsName = nil case, are you also
recommending we keep the current:

  If `cgroupsName` is not specified, implementations can define the
  default cgroup path.

instead of the proposed “optional cgroups [and error if the runtime
choses not to use cgroups]” (keeping the current logic makes more
sense to me [1])?

[1]: https://github.com/opencontainers/specs/pull/270#discussion_r47825968